pkg/controller/deployment/deployment_controller.go

```diff
@@ -340,6 +396,10 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 
+	if dc.garbageCollectorEnabled {
```

Comment:
Note that listing all RS in the namespace for every D for every resync is going to be fairly expensive.
cc: @smarterclayton